polyfill/lib/calendar.mjs

```diff
@@ -1408,7 +1408,9 @@ const makeHelperGregorian = (id, originalEras) => {
           currentValue != value &&
           !ES.Call(ArrayPrototypeIncludes, aliases || [], [currentValue])
         ) {
-          throw new RangeErrorCtor(`Input ${name} ${currentValue} doesn't match calculated value ${value}`);
+          // Prefer displaying an era alias, instead of "gregory-inverse"
+          const expected = aliases?.[0] ?? value;
+          throw new RangeErrorCtor(`Input ${name} ${currentValue} doesn't match calculated value ${expected}`);
```

Comment:
I don't know about this change, though... it's weird to prefer alias data over primary, and I'm not sure it helps things outside of the Gregorian error messages (e.g., "gregory" vs. "gregory-inverse" for the Japanese calendar is probably worse than "japanese" vs. "japanese-inverse"). If we really want improvement here, it should probably take the form of updating the Era data model to separate `code` vs. `name` concepts, using the latter ("ethiopic-amete-alem", "{,b}ce", etc.) in error messages:
```js
/**
 * This function adds additional metadata that makes it easier to work with
 * eras. Note that it mutates and normalizes the original era objects, which is
 * OK because this is non-observable, internal-only metadata.
 * For details on codes and names, see
 * https://tc39.es/proposal-intl-era-monthcode/#table-eras
 *
 *  interface Era {
 *   // Code is used to populate the "era" field of Temporal instances.
 *   code: string;
 *
 *   // Names are accepted as input, and the first name is also output in error
 *   // messages.
 *   // If absent, this field defaults to a single element matching the code.
 *   names?: string[];
 *
 *   …
 *  }
 */
```
```js
const helperCoptic = makeHelperOrthodox('coptic', [
  { code: 'coptic', isoEpoch: { year: 284, month: 8, day: 29 } },
  { code: 'coptic-inverse', reverseOf: 'coptic' }
]);
// Anchor is currently the older era to match ethioaa, but should it be the newer era?
// See https://github.com/tc39/ecma402/issues/534 for discussion.
const helperEthiopic = makeHelperOrthodox('ethiopic', [
  { code: 'ethioaa', names: ['ethiopic-amete-alem', 'mundi'], isoEpoch: { year: -5492, month: 7, day: 17 } },
  { code: 'ethiopic', names: ['incar'], isoEpoch: { year: 8, month: 8, day: 27 }, anchorEpoch: { year: 5501 } }
]);
…
const helperGregory = ObjectAssign(
  makeHelperSameMonthDayAsGregorian('gregory', [
    { code: 'gregory', names: ['ce', 'ad'], isoEpoch: { year: 1, month: 1, day: 1 } },
    { code: 'gregory-inverse', names: ['bce', 'bc'], reverseOf: 'gregory' }
  ]),
  …
);
…
  makeHelperSameMonthDayAsGregorian('japanese', [
    // The Japanese calendar `year` is just the ISO year, because (unlike other
    // ICU calendars) there's no obvious "default era", we use the ISO year.
    { code: 'reiwa', isoEpoch: { year: 2019, month: 5, day: 1 }, anchorEpoch: { year: 2019, month: 5, day: 1 } },
    { code: 'heisei', isoEpoch: { year: 1989, month: 1, day: 8 }, anchorEpoch: { year: 1989, month: 1, day: 8 } },
    { code: 'showa', isoEpoch: { year: 1926, month: 12, day: 25 }, anchorEpoch: { year: 1926, month: 12, day: 25 } },
    { code: 'taisho', isoEpoch: { year: 1912, month: 7, day: 30 }, anchorEpoch: { year: 1912, month: 7, day: 30 } },
    { code: 'meiji', isoEpoch: { year: 1868, month: 9, day: 8 }, anchorEpoch: { year: 1868, month: 9, day: 8 } },
    { code: 'japanese', names: ['japanese', 'gregory', 'ad', 'ce'], isoEpoch: { year: 1, month: 1, day: 1 } },
    { code: 'japanese-inverse', names: ['japanese-inverse', 'gregory-inverse', 'bc', 'bce'], reverseOf: 'japanese' }
  ]),
```

Reply:
I like both of these suggestions. Thanks!